### PR TITLE
[client] Improve route acl

### PR DESCRIPTION
**Pull request by lixmal**
## Describe your changes

- Updates nftables library to v0.2.0
- Marks traffic that was originally destined for local and applies the input rules in the forward chain if said traffic was redirected (e.g. by Docker)
- Adds nft rules to internal map only if flush was successful
- Improves error message if handle is 0 (= not found or hasn't been refreshed)
- Adds debug logging when route rules are added
- Replaces nftables userdata (rule ID) with rule hash

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/firewall/iptables/acl_linux.go [hunk 1]
```diff
@@ -232,6 +233,23 @@ func (m *aclManager) cleanChains() error {
...
+		return fmt.Errorf("list chains: %w", err)
+	}
+	if ok {
+		for _, rule := range m.entries["PREROUTING"] {
+			err := m.iptablesClient.DeleteIfExists("mangle", "PREROUTING", rule...)
+			if err != nil {
+				log.Errorf("failed to delete rule: %v, %s", rule, err)
+			}
+		}
+		err = m.iptablesClient.ClearChain("mangle", "PREROUTING")
```
Comment (mlsmaycon) [inline client/firewall/iptables/acl_linux.go]: This is cleaning any rules in the chain.

File: client/firewall/iptables/acl_linux.go [hunk 2]
```diff
@@ -232,6 +233,23 @@ func (m *aclManager) cleanChains() error {
 		}
 	}
 
+	ok, err = m.iptablesClient.ChainExists("mangle", "PREROUTING")
```
Comment (mlsmaycon) [inline client/firewall/iptables/acl_linux.go:244]: should it be mangle or nat?

File: client/firewall/iptables/acl_linux.go [hunk 3]
```diff
@@ -292,7 +310,10 @@ func (m *aclManager) seedInitialEntries() {
 
 	m.appendToEntries("FORWARD", []string{"-i", m.wgIface.Name(), "-j", "DROP"})
 	m.appendToEntries("FORWARD", []string{"-i", m.wgIface.Name(), "-j", m.routingFwChainName})
+	m.appendToEntries("FORWARD", []string{"-m", "mark", "--mark", fmt.Sprintf("%#x", nbnet.PreroutingFwmark), "-j", chainNameInputRules})
 	m.appendToEntries("FORWARD", append([]string{"-o", m.wgIface.Name()}, established...))
+
+	m.appendToEntries("PREROUTING", []string{"-t", "nat", "-i", m.wgIface.Name(), "-m", "addrtype", "--dst-type", "LOCAL", "-j", "MARK", "--set-mark", fmt.Sprintf("%#x", nbnet.PreroutingFwmark)})
```
Comment (mlsmaycon) [inline client/firewall/iptables/acl_linux.go]: nat or mangle?

**Review by mlsmaycon — CHANGES_REQUESTED** [review]

File: client/firewall/iptables/acl_linux.go (hunk 1 above)
Comment (lixmal) [reply client/firewall/iptables/acl_linux.go]: Indeed, see the comment on top of the fn

```
// todo write less destructive cleanup mechanism
```

I've restored the previous code when we still had prerouting rules.

File: client/firewall/iptables/acl_linux.go (hunk 2 above)
Comment (lixmal) [reply client/firewall/iptables/acl_linux.go:244]: mangle

File: client/firewall/iptables/acl_linux.go (hunk 3 above)
Comment (lixmal) [reply client/firewall/iptables/acl_linux.go]: mangle, fixed

File: client/firewall/iptables/acl_linux.go (hunk 1 above)
Comment (mlsmaycon) [reply client/firewall/iptables/acl_linux.go]: let's not keep an old bug

Comment (lixmal) [reply client/firewall/iptables/acl_linux.go]: fixed

**Pull request merged**